### PR TITLE
Update XML editor content after changes

### PR DESCRIPTION
**Pull request by reckseba**
RISDEV-4191

### CODE REVIEW

**Comment by reckseba** [comment]
@malte-laukoetter would you say that this is a valid attempt? Let's discuss. If it's fine I'd provide the tests.

**Comment by malte-laukoetter** [comment]
> @malte-laukoetter would you say that this is a valid attempt? Let's discuss. If it's fine I'd provide the tests.

yes looks like a good solution, I'm fine with reusing the endpoint for the html render also for the — let's call it xml-render — of the norm at a specific point in time ^^

File: frontend/src/composables/useNormRender.ts [hunk 1]
```diff
@@ -7,12 +7,14 @@ import { UseFetchReturn } from "@vueuse/core"
...
  * @param normXml XML of the norm that should be rendered
  * @param showMetadata Enable or disable metadata list at the beginning of the document
+ * @param renderHtml By default (true) renders HTML. When false then XML.
  * @param customNorms The XMLs of norms which are referenced by the norm (e.g. in passiveModifications) and should be used instead of the data stored.
  * @param at Passive modifications coming into effect before this date should be applied before rendering the HTML
  */
 export function useNormRender(
   normXml: MaybeRefOrGetter<string | undefined>,
   showMetadata: MaybeRefOrGetter<boolean> = false,
+  renderHtml: MaybeRefOrGetter<boolean> = true,
```
Comment (malte-laukoetter) [inline frontend/src/composables/useNormRender.ts]: I'm unsure if this additional flag is the best solution here - or if we want to just include a second method. Also with this many parameters (and multiple once that are booleans) I would find it nice if we could move to an options object for all the optional parameters.

Comment (reckseba) [reply frontend/src/composables/useNormRender.ts]: Alright! I wavered between the two options. I removed the parameter and introduced a second method instead.

If this is fine, I'll provide the missing FE & BE tests on Monday.

**Comment by reckseba** [comment]
@malte-laukoetter I provided the necessary tests. Though the e2e test is failing and I actually have no clue why. Do you?

**Comment by malte-laukoetter** [comment]
> @malte-laukoetter I provided the necessary test. Though the e2e test is failing and I actually have no clue why. Do you?

The xml editor is using virtual scrolling to not need to render the complete xml all the time. Therefore the new text might not be rendered on the page and can therefore not be found by playwright. I don't think i ever found a good solution to that problem :/ Maybe you have an idea?

**Review by malte-laukoetter — APPROVED** [review]

**Comment by reckseba** [comment]
> > @malte-laukoetter I provided the necessary test. Though the e2e test is failing and I actually have no clue why. Do you?
> 
> The xml editor is using virtual scrolling to not need to render the complete xml all the time. Therefore the new text might not be rendered on the page and can therefore not be found by playwright. I don't think i ever found a good solution to that problem :/ Maybe you have an idea?

That's why.. You're right. Thanks! 

And yes. I had and idea: There is this search feature. I used that one. Good?

**Pull request merged**